include/plugin_interface/SC_Unit.h

```diff
@@ -80,6 +80,17 @@ enum { kUnitDef_CantAliasInputsToOutputs = 1 };
 // set the calculation function
 #define SETCALC(func) (unit->mCalcFunc = (UnitCalcFunc)&func)
 
+#define ClearUnitOnMemFailed                                                                                           \
+    Print("%s: alloc failed, increase server's RT memory (e.g. via ServerOptions)\n", __func__);                       \
+    SETCALC(*ClearUnitOutputs);                                                                                        \
+    unit->mDone = true;                                                                                                \
+    return;
+
+#define ClearUnitIfMemFailed(ptr)                                                                                      \
+    if (ptr == nullptr) {                                                                                              \
+        ClearUnitOnMemFailed                                                                                           \
+    }
```

Comment:
The macro expects a pointer, but you sometimes use it with a boolean expression, e.g. https://github.com/supercollider/supercollider/pull/5713/files#diff-06296acaab6425bef65d947ed70f147408a421060d32ed7601c9b18654bdf802R143
You shouldn't really compare a `bool` with `nullptr`.

I think it's better to just take a boolean expression instead of a pointer, so it works cleanly with both:
```
#define ClearUnitIfMemFailed(condition)   \
    if (!condition) {                     \
        ClearUnitOnMemFailed              \
    }
```

Reply:
Good point, thanks, and I also see we need parenthesis around `ptr` (or `condition`) in the `if` check.

I'm afraid my next question here can be a bit bikeshedding, but I'd like to evaluate two options here, to make this cleaner:
1. Rename `ClearUnitIfMemFailed` to something like `ClearUnitIfConditionFailed`, `ClearUnitIfFalse`, or `CheckMemOrClearUnit`, AssertOrClearUnit`, or anything else that would better describe that we just check a boolean condition there, and if it's false we consider it an allocation failure and clear the unit accordingly.
2. Leave the `if` to the user, and only introduce `ClearUnitOnMemFailed`. The drawback is that we loose authority on the if check.

Reply:
Ad 1: Since `ClearUnitIfMemFailed` calls `ClearUnitOnMemFailed` under the hood and the latter prints an error message about the RT memory pool, it's good to keep the `MemFailed` part and not use something generic like `ClearUnitIfFalse`.

Ad 2: First I thought the macro wouldn't be necessary, but it actually is quite handy.

Reply:
Great, I'll just fix the parenthesis then and leave the name as it is!

Reply:
@Spacechild1 - do you feel this is resolved now? Looks right to me.

Reply:
Yep, looks good! Thanks for the heads-up.